### PR TITLE
[New Term] C++ Math Functions: `round()`

### LINKED ISSUE #986: [Term Entry] C++ Math Functions: `round()`
### Reason/inspiration (optional)

Under c++/concepts/math-functions/terms/, create a new term entry for the `round()` function.

In the terms/ folder, make a new folder called `round`
Inside the isnan folder, create a new markdown file called `round.md`
Inside the file, be sure to include a "Syntax" section with a \```pseudo code snippet.
Let's include an "Example" section with a \```cpp code snippet.
Lastly, let's include a "Codebyte Example" with a runnable \```codebyte/cpp
Please refer to the [entry templates](https://github.com/Codecademy/docs/blob/main/documentation/), [content standards](https://github.com/Codecademy/docs/blob/main/documentation/content-standards.md), and [markdown style guide](https://curriculum-documentation.codecademy.com/content-guidelines/markdown-style-guide/) when working on the PR for this issue.

### Entry Type (select all that apply)

New Term for Existing Concept Entry

### Code of Conduct

- [X] I agree to follow Codecademy Doc's Code of Conduct

### For Maintainers

- [X] Labels added
- [ ] Issue is assigned

### PR DESCRIPTION
**Pull request by tanishq-singh-2301**
### Description

Closes #986

### Type of Change

- Adding a new entry

### Checklist

- [x] All writings are my own.
- [x] My entry follows the Codecademy Docs style guide.
- [x] My changes generate no new warnings.
- [x] I have performed a self-review of my own writing and code.
- [x] I have checked my entry and corrected any misspellings.
- [x] I have made corresponding changes to the documentation if needed.
- [x] I have confirmed my changes are not being pushed from my forked `main` branch.
- [x] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [x] Under "Development" on the right, I have linked any issues that are relevant to this PR (write "Closes #<issue number> in the "Description" above).

### CODE REVIEW

**Comment by Dusch4593** [comment]
@tanishq-singh-2301 Thanks for doing that, much appreciated! 😄 After I'm finished with review I'll tag you again so you can implement the feedback.

**Comment by tanishq-singh-2301** [comment]
Sure,
Thanks.

File: content/cpp/concepts/math-functions/terms/round/round.md
Comment (Dusch4593) [inline content/cpp/concepts/math-functions/terms/round/round.md:15]:
```suggestion
The `round()` function returns the integer that is nearest to the argument, with halfway cases rounded away from zero.
```

File: content/cpp/concepts/math-functions/terms/round/round.md
Comment (Dusch4593) [inline content/cpp/concepts/math-functions/terms/round/round.md:3]:
```suggestion
Description: 'Returns the integer that is nearest to the argument, with halfway cases rounded away from zero.'
```

File: content/cpp/concepts/math-functions/terms/round/round.md
Comment (Dusch4593) [inline content/cpp/concepts/math-functions/terms/round/round.md:20]:
```suggestion

```

I would omit this line since this point is covered in the parent [math functions entry](https://www.codecademy.com/resources/docs/cpp/math-functions).

File: content/cpp/concepts/math-functions/terms/round/round.md
Comment (Dusch4593) [inline content/cpp/concepts/math-functions/terms/round/round.md:22]:
```suggestion
round(num);
```

File: content/cpp/concepts/math-functions/terms/round/round.md
Comment (Dusch4593) [inline content/cpp/concepts/math-functions/terms/round/round.md:25]:
```suggestion
The `num` parameter must be a `double`,`float`, or `long double`. The return value will be an integer.
```

File: content/cpp/concepts/math-functions/terms/round/round.md
Comment (Dusch4593) [inline content/cpp/concepts/math-functions/terms/round/round.md:34]: This example could be expanded a bit more. Can we assign a numeric value to `num` before passing into `round()`?

**Review by Dusch4593 — CHANGES_REQUESTED** [review]
@tanishq-singh-2301 Left you some comments/suggestions to look at. Tag when when you're finished with implementing the feedback and I'll take another look. 😄

**Comment by tanishq-singh-2301** [comment]
Done.
Please review updated code @Dusch4593 

Thanks

File: content/cpp/concepts/math-functions/terms/round/round.md
Comment (Dusch4593) [inline content/cpp/concepts/math-functions/terms/round/round.md:15]:
```suggestion
The **`round()`** function returns the integer that closest to the argument, with halfway cases rounded away from zero.
```

File: content/cpp/concepts/math-functions/terms/round/round.md
Comment (Dusch4593) [inline content/cpp/concepts/math-functions/terms/round/round.md:23]: I would maybe include one more sentence here that explains what a halfway case is and how they are used in the `round()` function.

File: content/cpp/concepts/math-functions/terms/round/round.md
Comment (Dusch4593) [inline content/cpp/concepts/math-functions/terms/round/round.md:39]:
```suggestion
The following example is runnable and rounds the halfway case away from zero:
```

File: content/cpp/concepts/math-functions/terms/round/round.md
Comment (Dusch4593) [inline content/cpp/concepts/math-functions/terms/round/round.md:26]: Let's add one sentence there that previews the \```cpp snippet. Maybe something that starts with "The following example"

**Review by Dusch4593 — CHANGES_REQUESTED** [review]
@tanishq-singh-2301 Just a few more comments from me 😄

**Comment by tanishq-singh-2301** [comment]
Done.
Please review updated code @Dusch4593 

Thanks

File: content/cpp/concepts/math-functions/terms/round/round.md
Comment (Dusch4593) [inline content/cpp/concepts/math-functions/terms/round/round.md:24]:
```suggestion
The `num` parameter must be a `double`,`float`, or `long double`. The return value will be an integer. If `num` is a halfway case where the decimal place is `.5` or higher, then the closest integer above `num` is returned.
```

File: content/cpp/concepts/math-functions/terms/round/round.md
Comment (Dusch4593) [inline content/cpp/concepts/math-functions/terms/round/round.md:49]:
```suggestion
#include <iostream>
#include <cmath>

int main() {
  double num1 = 9.23;
  double result1;
  result1 = std::round(num1);

  std::cout << "The result of round(9.23) is " << result1 << "!\n";

  double num2 = 4.5;
  double result2;
  result2 = std::round(num2);

  std::cout << "The result of round(4.5) is " << result2 << "!\n";
}
```

File: content/cpp/concepts/math-functions/terms/round/round.md
Comment (Dusch4593) [inline content/cpp/concepts/math-functions/terms/round/round.md:39]: Let's keep the entry to just one "Example" and one "Codebyte Example". My suggestion here would be to combine the code from both of these snippets.

```suggestion
```

File: content/cpp/concepts/math-functions/terms/round/round.md
Comment (Dusch4593) [inline content/cpp/concepts/math-functions/terms/round/round.md:40]:
```suggestion
## Example
```

File: content/cpp/concepts/math-functions/terms/round/round.md
Comment (Dusch4593) [reply content/cpp/concepts/math-functions/terms/round/round.md:41]:
```suggestion

The following example showcases the `round()` function being applied to two `double` values, one of which is a halfway case:

```

File: content/cpp/concepts/math-functions/terms/round/round.md
Comment (Dusch4593) [reply content/cpp/concepts/math-functions/terms/round/round.md:49]: I think we should include the `#includes` statements along with the code being nested in a `main()` method so that if a reader copy/pastes this example they can run it. 😄

File: content/cpp/concepts/math-functions/terms/round/round.md
Comment (Dusch4593) [inline content/cpp/concepts/math-functions/terms/round/round.md:67]:
```suggestion
  std::cout << "The result is " << result << "!\n";
```

Comment (Dusch4593) [reply content/cpp/concepts/math-functions/terms/round/round.md:67]: No need for output comments in Codebyte snippets.

File: content/cpp/concepts/math-functions/terms/round/round.md
Comment (Dusch4593) [inline content/cpp/concepts/math-functions/terms/round/round.md:51]:
```suggestion

This produces the following output:

```

Then add a new \```shell code snippet with the following:

```shell
The result of round(9.23) is 9!
The result of round(4.5) is 5!
```

File: content/cpp/concepts/math-functions/terms/round/round.md
Comment (Dusch4593) [inline content/cpp/concepts/math-functions/terms/round/round.md:3]:
```suggestion
Description: 'Returns the integer that is nearest to the argument, with halfway cases rounded away from the ending zero.'
```

File: content/cpp/concepts/math-functions/terms/round/round.md
Comment (Dusch4593) [inline content/cpp/concepts/math-functions/terms/round/round.md:15]:
```suggestion
The **`round()`** function returns the integer that is closest to the argument, with halfway cases rounded away from the ending zero.
```

**Comment by tanishq-singh-2301** [comment]
working on it!

**Comment by tanishq-singh-2301** [comment]
Done, please review and tell me if any more changes is required?
@Dusch4593 

Thanks

**Review by Dusch4593 — APPROVED** [review]
@tanishq-singh-2301 LGTM for the next review! 👍🏻

**Comment by tanishq-singh-2301** [comment]
Any update? @Dusch4593

**Comment by Dusch4593** [comment]
@tanishq-singh-2301 Since I've already approved changes and the first review is finished, someone else from my team will step in (most likely this week) to conduct a second review. 😄

File: content/cpp/concepts/math-functions/terms/round/round.md
Comment (HishamT) [inline content/cpp/concepts/math-functions/terms/round/round.md:23]: ~If `num` is a halfway case where the decimal place is `.5` or higher, then the closest integer above `num` is returned.~

Consider this instead:

"If the decimal in `num` is 0.5 or higher, the closest integer greater than `num` is returned."

File: content/cpp/concepts/math-functions/terms/round/round.md
Comment (HishamT) [inline content/cpp/concepts/math-functions/terms/round/round.md:38]: Remove the exclamation mark ('!').

File: content/cpp/concepts/math-functions/terms/round/round.md
Comment (HishamT) [inline content/cpp/concepts/math-functions/terms/round/round.md:44]: Remove the exclamation mark ('!').

**Review by HishamT — CHANGES_REQUESTED** [review]
Finished reviewing the entry. Left some suggestions.

**Review by HishamT — APPROVED** [review]
LGTM!

**Comment by tanishq-singh-2301** [comment]
Okay

**Comment by tanishq-singh-2301** [comment]
any update? @HishamT

**Review by HishamT — APPROVED** [review]
LGTM!

**Review by HishamT — APPROVED** [review]

File: content/cpp/concepts/math-functions/terms/round/round.md
Comment (yangc95) [reply content/cpp/concepts/math-functions/terms/round/round.md:15]: @tanishq-singh-2301 Let's say something along this lines of he number being rounded up. More straightforward this way.

**Review by yangc95 — COMMENTED** [review]
I had one additional comment before merge.

**Review by yangc95 — APPROVED** [review]
LGTM

**Pull request merged**
